graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,273 @@
+            if np.ndim(X) != 2:
+                raise IndexError("Argument must have dim 2")
+            if X.shape[0] != X.shape[1]:
+                raise IndexError("Adjacency Matrix should be square.")
+        else:
+            raise NotImplementedError("Multigraph SVN not implemented")
+
+        # Embed graph if embedding not provided
+        if self.embedding is None:
+            if issubclass(type(self.embedder), BaseSpectralEmbed):
+                self.embedding = self.embedder.fit_transform(X)
+            else:
+                raise TypeError("No embedder available")
+
+    def _predict(self, k: int = 5) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Nominate vertex based on attribute specific distance from the k nearest neighbors.
+
+        Parameters
+        ----------
+        k : int
+            Number of neighbors to consider in nearest neighbors distance weighting.
+
+        Returns
+        -------
+        An tuple of two np.ndarrays, each with shape = ``(number_vertices, number_attributes_in_seed)``.
+        The array at index 0 is the nomination list. For each attribute column, the rows contain indices
+        of vertices in original adjacency matrix ordered by likelihood of matching that attribute.
+        The array at index 1 is the computed distances, where each element at (i, j) represents the
+        distance metric value between vertex i and attribute j.
+
+        """
+        num_unique = self.unique_att_.shape[0]
+        ordered = self.distance_matrix_.argsort(axis=1)
+        sorted_dists = self.distance_matrix_[np.arange(ordered.shape[0]), ordered.T].T
+        nd_buffer = np.tile(
+            self.attr_labels_[ordered[:, :k]], (num_unique, 1, 1)
+        ).astype(np.float64)
+
+        # comparison taking place in 3-dim view, coordinates produced are therefore 3D
+        inds = np.argwhere(nd_buffer == self.unique_att_[:, np.newaxis, np.newaxis])
+
+        # nans are a neat way to operate on attributes individually
+        nd_buffer[:] = np.NaN
+        nd_buffer[inds[:, 0], inds[:, 1], inds[:, 2]] = sorted_dists[
+            inds[:, 1], inds[:, 2]
+        ]
+
+        # weighting function. Outer inverse for consistency, makes equivalent to simple
+        # ranking by distance in unattributed case, and makes higher ranked vertices
+        # naturally have lower distance metric value.
+        pred_weights = np.power(np.nansum(np.power(nd_buffer, -1), axis=2), -1).T
+
+        nan_inds = np.argwhere(np.isnan(pred_weights))
+        pred_weights[nan_inds[:, 0], nan_inds[:, 1]] = np.inf
+        vert_order = np.argsort(pred_weights, axis=0)
+
+        inds = np.tile(self.unique_att_, (1, vert_order.shape[0])).T
+        inds = np.concatenate((vert_order.reshape(-1, 1), inds), axis=1)
+        pred_weights = pred_weights[inds[:, 0], inds[:, 1]]
+        return vert_order, pred_weights.reshape(vert_order.shape)
+
+    def predict(self, k: int = None) -> Tuple[np.ndarray, np.ndarray]:
```

Comment:
I only allow euclidean now because I haven't tested other distance metrics. I think it could be useful to add mahalanobis distance and maybe a couple others, but that exposing all the options available from scipy's `cdist` wouldn't be very useful, and could possibly be confusing

Reply:
i think then you should add a `metric` parameter with euclidean (default) and mahalanabis as options, and we can add more in the future